src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt

```diff
@@ -32,6 +32,27 @@ val RsQualifiedNamedElement.qualifiedName: String?
         return "$cargoTarget$inCratePath"
     }
 
+fun RsQualifiedNamedElement.qualifiedNameRelativeTo(mod: RsMod): String? {
+    val crateRelativePath = crateRelativePath
+    if (mod.crateRoot != crateRoot || crateRelativePath == null) {
+        return qualifiedName
+    }
+
+    val modPathPrefix = mod.crateRelativePath?.let { "$it::" }
+    return if (containingMod.superMods.contains(mod)
+        && modPathPrefix != null
+        && crateRelativePath.startsWith(modPathPrefix)
+    ) {
+        var result = crateRelativePath.removePrefix(modPathPrefix)
+        if (cargoWorkspace?.packages?.any { result.startsWith("${it.normName}::") } == true) {
```

Comment:
Let's introduce `cargoWorkspace` not to deal with nullability and `== true`. And probably it can be `val cargoWorkspace = ... ?: return ...`, but you can choose any way of doing this